### DISCUSSION TITLE
Hi, I'm not able to install NextAuth in Step 6 of the tutorial.

### DISCUSSION THREAD

**Discussion by mikedaltonmtl** [prisma/prisma#18377] [Q&A] [answered]
The error message is as follows:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: hello-next@1.0.0
npm ERR! Found: next@12.0.10
npm ERR! node_modules/next
npm ERR!   next@"12.0.10" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer next@"^12.2.5 || ^13" from next-auth@4.20.1
npm ERR! node_modules/next-auth
npm ERR!   next-auth@"4" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```
Does this mean that I need to upgrade my version of Next.js?
If so, am I able to do this without disrupting the project so far?

Any help would be greatly appreciated!

**Comment by ludralph** [comment, marked as answer]
Hi @mikedaltonmtl 👋 

You have Next.js version 12.0.10 installed in your project. However, next-auth requires a version of Next.js that satisfies the following peer dependency requirement: ^12.2.5 || ^13.

One solution to this problem would be to upgrade your version of Next.js to a version that satisfies the peer dependency requirement of next-auth.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise, please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by mikedaltonmtl** [reply]
Thank you so much for your help Raphael, I'm new to this and didn't expect the fix to be so simple!

If anyone else has the same problem, what worked for me was to update React first:
```
npm install react@latest react-dom@latest
```
and then update Next.js
```
npm i next@latest react@latest react-dom@latest eslint-config-next@latest
```
and lastly, there was a Link component to fix (remove the embedded <a> tag).
This document has the details: https://nextjs.org/docs/upgrading

Thanks again for the help!

**Reply by ludralph** [reply]
Hi @mikedaltonmtl, I'm happy to help.
